http/base/src/main/java/org/wildfly/security/http/HttpAuthenticator.java

```diff
@@ -311,6 +308,10 @@ private class AuthenticationExchange extends BaseHttpServerRequest implements Ht
             super(httpExchangeSpi);
         }
 
+        private boolean isAuthenticated() {
+            return authenticated;
```

Comment:
@keshav-redhat Thank you for the PR! 

Just a minor, it seems that the variable `authenticated` can be moved to the `AuthenticationExchange` class as well